src/route/user.route.js

```diff
@@ -0,0 +1,42 @@
+import userController from '../controller/user.controller.js'
+import express from 'express'
+import schema from './user.post.schema'
+
+const userRouter = express.Router();
+
+userRouter.post('/add', validateSchema(schema), userController.addUser);
+userRouter.get('/all', userController.getAllUsers);
+userRouter.get('/:id', userController.getUserById);
+userRouter.get('/:login/:limit', userController.getAutoSuggestUsers);
+userRouter.put('/:id', validateSchema(schema), userController.updateUser);
+userRouter.delete('/:id', userController.deleteUser);
+
+
+function errorResponse(schemaErrors) {
+    const errors = schemaErrors.map((error) => {
+        let {path, message} = error;
+        return {path, message};
+    });
+
+    return {
+        status: 'failed',
+        errors,
+    };
+}
+
+function validateSchema(schema) {
+    return (request, response, next) => {
+        const {error} = schema.validate(request.body, {
+            abortEarly: false,
+            allowUnknown: false,
+        });
+
+        if (error !== undefined && error.isJoi) {
```

Comment:
use short syntax to check if error is defined, just `if (error){...}`
the second condition is redundant